src/odemis/gui/cont/tabs.py

```diff
@@ -4455,7 +4469,496 @@ def _on_align_move_done(self, f=None):
+                        FIB_line_overlay.shape = HORIZONTAL_LINE
+                        break
+
+        doc_path = pkg_resources.resource_filename("odemis.gui", "doc/enzel_z_alignment.html")
+        self.panel.html_alignment_doc.LoadPage(doc_path)
+
+        self.panel.controls_step_size_slider.SetRange(100e-9, 10e-6)
+        self.panel.controls_step_size_slider.set_position_value(100e-9)
+
+        self.panel.pnl_z_align_controls.Show(True)
+        self.panel.pnl_sem_align_controls.Show(False)
+        self.panel.pnl_flm_align_controls.Show(False)
+        fix_static_text_clipping(self.panel)
+        self.tab_data_model.align_mode.value = "Z alignment"
+
+    def _z_alignment_controls(self, axis, proportion, evt):
+        """
+        Links the Z alignment mode control buttons to the stage and updates the correct metadata.
+
+        :param axis (str): axis which is controlled by the button.
+        :param proportion (int): direction of the movement represented by the button (-1/1)
+        :param evt (GenButtonEvent): Clicked event
+        """
+        # Only proceed if there is no currently running target_position
+        if hasattr(self, "_z_move_future") and self._z_move_future._state == RUNNING:
+            logging.debug("The stage is still moving, this movement isn't performed.")
+            return
+
+        if hasattr(self, "_z_move_future") and not self._z_move_future.button.Enabled:
+            logging.error("The stream is still updating and hence button activity is suppressed.")
+            return
+        evt.theButton.Enabled = False  # Disable the button to prevent queuing of movements and unnecessary refreshing.
+
+        self._z_move_future = self._stage_global.moveRel(
+                {axis: proportion * self.tab_data_model.step_size.value})
+        self._z_move_future.button = evt.theButton
+        self._z_move_future.add_done_callback(self._z_alignment_move_done)
+
+    @call_in_wx_main
+    def _z_alignment_move_done(self, future):
+        self.stream_bar_controller.refreshStreams((self._fib_stream,
+                                                   self._sem_stream))
+        # Save the new position in the metadata
+        self._stage.updateMetadata({model.MD_FAV_POS_ACTIVE: self._stage.position.value})
+        future.button.Enabled = True
+
+    def _set_sem_alignment_mode(self):
+        """
+        Sets the SEM alignment mode with the correct streams, stream controllers, instructions and alignment controls.
+        """
+        self.stream_bar_controller.pauseAllStreams()
+        self._set_top_and_bottom_stream_and_settings(self._SEM_view_and_control,
+                                                     self._FIB_view_and_control)
+
+        # Adjust the overlay to a normal crosshair in the FIB viewport
+        for viewport in self.panel.pnl_two_streams_grid.viewports:
+            if viewport.view.stream_classes is acqstream.FIBStream:
```

Comment:
That feels a little indirect.
You could write `fib_viewport = self._FIB_view_and_control["viewport"]`, and you'd be done.